feat: allow number cards in workspaces

Alchez opened the pull request:
https://user-images.githubusercontent.com/30859809/221816913-53676af0-f5c6-4fcf-9b75-4e0b0ae3d1fd.mov

<hr>

[docs](https://frappeframework.com/docs/v14/user/en/desk/workspace/blocks/edit-wiki?wiki_page_patch=592cbb4c18)

>no-docs

Commit:
feat: allow number cards in workspaces

Alchez commented:
bump

shariquerik committed:
chore: code reformatting

shariquerik committed:
fix: added shadow for cards only on workspace

shariquerik committed:
fix: using base_width set_title logic

shariquerik committed:
fix: keep number_card action only in readonly mode (not in customize mode)

shariquerik committed:
fix: only allow to add existing number card on workspace

separated logic for dashboard and workspace implementation

shariquerik committed:
Merge branch 'develop' into feat-number-card-workspace-dev

shariquerik committed:
fix: added number_card icon

shariquerik committed:
fix: prioritize workspace_number_card doctype while installing

shariquerik committed:
fix: include number_card in cleanup logic

Alchez commented:
@shariquerik, thanks for the changes! Let me know when you're done with them, and I can look into adding to the [documentation](https://frappeframework.com/docs/v14/user/en/desk/workspace/blocks).

shariquerik commented:
> @shariquerik, thanks for the changes! Let me know when you're done with them, and I can look into adding to the [documentation](https://frappeframework.com/docs/v14/user/en/desk/workspace/blocks).

@Alchez It's done, can you also add a UI test?

shariquerik commented:
> @Alchez It's done, can you also add a UI test?

@Alchez Are you working on the UI test? Or should I add it?

Alchez commented:
@shariquerik, I got busy with other work, but I can come back to this later. I haven't written a UI test before, so I'm okay if you write one for this or point me to the right place to start writing one.

shariquerik commented:
> @shariquerik, I got busy with other work, but I can come back to this later. I haven't written a UI test before, so I'm okay if you write one for this or point me to the right place to start writing one.

I will write for this, but you can learn Cypress UI test later for future contributions, there are many examples in the frappe repository. Cypress documentation will be enough to start learning.
Command to start the Cypress test in local

```
bench --site <site-name> run-ui-tests frappe
```

Alchez commented:
@shariquerik, alright thanks! I've added documentation (atleast a patch request) for this PR. I didn't find a public link to share the contribution, but I've added the link in the PR description anyway since it contains the ID for the patch.

shariquerik committed:
test: UI test for number_card workspace block

shariquerik committed:
Merge branch 'develop' into feat-number-card-workspace-dev
